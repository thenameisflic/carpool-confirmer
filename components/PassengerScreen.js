import React from "react";
import { View, StyleSheet, Button } from "react-native";
import Title from "./typography/Title";
import theme from "../theme";
import RoleIcon from "./RoleIcon";
import RoleToggler from "./RoleToggler";
import QRCodeScanner from "react-native-qrcode-scanner";
import { ROLES } from "../constants";
import { createResetAction } from "../utils";

export default function PassengerScreen({navigation}) {
  const onQrCode = e => {
    console.log(e.data);
  };

  return (
    <View style={styles.container}>
      <Button
        title="TODO: Skip"
        onPress={() =>
          navigation.dispatch(
            createResetAction({
              routeName: "ShareLocation",
              params: { role: ROLES.PASSENGER }
            })
          )
        }
      />
      <Title size="h5" style={styles.title}>
        Escaneie o c√≥digo do seu motorista
      </Title>
      <QRCodeScanner onRead={onQrCode} />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: theme.backgroundColor,
    paddingHorizontal: 16,
    alignItems: "center",
    justifyContent: "center",
    flex: 1
  },
  title: {
    textAlign: "center",
    marginTop: 40,
    marginBottom: 40
  },
  headerLeft: {
    marginLeft: 8
  },
  headerRight: {
    paddingRight: 8
  }
});

PassengerScreen.navigationOptions = ({ navigation }) => ({
  title: "Passageiro",
  headerLeft: <RoleIcon style={styles.headerLeft} />,
  headerRight: (
    <View style={styles.headerRight}>
      <RoleToggler
        style={styles.headerRight}
        onPress={() => navigation.navigate("Driver")}
      />
    </View>
  )
});
