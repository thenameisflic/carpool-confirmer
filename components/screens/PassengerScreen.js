import React from "react";
import { View, StyleSheet, Button } from "react-native";
import Title from "../typography/Title";
import theme from "../../theme";
import RoleIcon from "../ui/RoleIcon";
import RoleToggler from "../ui/RoleToggler";
import QRCodeScanner from "react-native-qrcode-scanner";
import { ROLES } from "../../constants";
import { createResetAction } from "../../utils";
import RNLocation from "react-native-location";
import { scanPassenger } from "../../api";

export default function PassengerScreen({ navigation }) {
  const onQrCode = async e => {
    RNLocation.requestPermission({
      ios: "whenInUse",
      android: {
        detail: "coarse",
        rationale: {
          title: "Nós precisamos de acesso à sua localização",
          message: "",
          buttonPositive: "OK",
          buttonNegative: "Cancelar"
        }
      }
    });
    RNLocation.configure({ distanceFilter: 0 });
    const location = await RNLocation.getLatestLocation({ timeout: 60000 });
    const position = `${location.latitude},${location.longitude}`;
    const response = await scanPassenger(e.data, position);
    const {
      id,
      driverPosition,
      passengerPosition,
      distance
    } = await response.json();

    navigation.dispatch(
      createResetAction({
        routeName: "ShareLocation",
        params: {
          role: ROLES.PASSENGER,
          driverPosition,
          passengerPosition,
          distance,
          rideId: id
        }
      })
    );
  };

  return (
    <View style={styles.container}>
      <QRCodeScanner onRead={onQrCode} />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: theme.backgroundColor,
    paddingHorizontal: 16,
    alignItems: "center",
    justifyContent: "center",
    flex: 1
  },
  title: {
    textAlign: "center",
    marginTop: 40,
    marginBottom: 40
  },
  headerLeft: {
    marginLeft: 8
  },
  headerRight: {
    paddingRight: 8
  }
});

PassengerScreen.navigationOptions = ({ navigation }) => ({
  title: "Passageiro",
  headerLeft: <RoleIcon style={styles.headerLeft} role={ROLES.PASSENGER} color="primary" />,
  headerRight: (
    <View style={styles.headerRight}>
      <RoleToggler
        style={styles.headerRight}
        onPress={() => navigation.navigate("Driver")}
        role={ROLES.DRIVER}
      />
    </View>
  )
});
