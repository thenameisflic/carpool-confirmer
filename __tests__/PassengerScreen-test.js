import "react-native";
import React from "react";
import PassengerScreen from "../components/screens/PassengerScreen";

import renderer from "react-test-renderer";

jest.mock("react-native-qrcode-scanner", () => "QRCodeScanner");
jest.mock("react-native-location", () => ({}));

it("renders PassengerScreen correctly", () => {
  const tree = renderer.create(<PassengerScreen navigation={{navigate: ""}} />);
  expect(tree.toJSON()).toMatchSnapshot();
});
