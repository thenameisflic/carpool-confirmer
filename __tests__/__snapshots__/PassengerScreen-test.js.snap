// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PassengerScreen correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 16,
    }
  }
>
  <QRCodeScanner
    onRead={[Function]}
  />
</View>
`;
